### ISSUE TITLE
Duplicate scrobbles if Goole Play Music is paused and resumed

### ISSUE THREAD

**Issue by fhimpe** [tgwizard/sls#177]
I listen to music with Google Play Music. When I stop listening, I pause Google Play Music, and I resume play-back later on. This seems to result in two or even three scrobbles of the song which was paying while it was paused.

Currently I see the same song being played 3 times in a row: once 5 hours ago,once 4 hours ago, and just now. I paused playback during that song 5 hours ago, and just resumed it. I have no clue how it thinks that I played that same song also 4 hours ago.

If this cannot be properly detected, at least SLS should have the option to filter any duplicate consecutive scrobbles of the same song. That will at least prevent my last.fm statistics from being totally wrong.

**Comment by HumbleBeeBumbleBee** [comment]
Triplicate #52 #80 #177
Potential solution. https://github.com/tgwizard/sls/pull/149 Check it out!

**Comment by HumbleBeeBumbleBee** [comment]
@dror3go
If you know how download and build the .apk
Please use this branch https://github.com/tgwizard/sls/tree/testing-149
Otherwise you can use the latest .apk here.
https://drive.google.com/file/d/0B9lPaMaVVtiJLVlLendjSmxRZDA/view

**Comment by sasavuk** [comment]
I've been using the latest apk and still have this problem. It's more severe than just pausing and resuming. Sometimes the app will scrobble a paused song 20 times throughout the day while Google Play Music isn't even running.

**Comment by HumbleBeeBumbleBee** [comment]
HI @sasavuk this problem is being looked at. #149 or some variation of it may solve this problem. I will put together a new branch to test it again and a new apk (if you need it).

**Comment by HumbleBeeBumbleBee** [comment]
@fhimpe  @sasavuk 
https://drive.google.com/file/d/0B9lPaMaVVtiJYTVVVDR4YkU1QUU/view?usp=sharing
This is a beta testing version. Let me know if it solves your problem.
It has a different key so you'll have to uninstall and setup SLS again.

**Comment by sasavuk** [comment]
No dice :(
Tons of repeated scrobbles throughout today and yesterday.

**Comment by HumbleBeeBumbleBee** [comment]
Thanks @sasavuk.
@metanota do you have any idea what could be causing this issue?
I have tested #149 on a Samsung Galaxy S2 LTE Android 4.1.2 (API 16) with latest version of:

Confirmed:
* MortPlayer
* VLC

Not Confirmed
* Google Play Music
* Stock Player

I also noticed that when a song is paused before the half way point, SLS scrobbles/caches the song.
@metanota I wonder if there is anyway we could really fix all track handling issues.
* #177
* #169 
* #155 
* #80
* #69

**Comment by fhimpe** [comment]
I'm using Google Play Music, and I have double scrobbles even at moments while Google Play Music is not playing.I think this happens if you pause playback in Google Play Music, and then later on Google Play Music is killed by Android's Low Memory Killer, but restarted immediately because it has a permanent notification active. In that case it scrobbles the song again, even though it is not playing. Or at least that's my theory about what could be happening.

I think there is only one good way to fix this: have an option in SLS to not scrobble subsequent, duplicate songs. If you enable this option, of course it will fail to scrobble if you really play the same song multiple times after each other, but at least in my case this is very rare and far less worse than wrongly scrobbling the same song over and over again, which ruins my last.fm statistics.

**Comment by jimfarrand** [comment]
I'm also hitting this problem, or something very similar (multiple phantom scrobbles of a track paused in Google Music).  fhimpe's diagnosis sounds pretty plausible to me.

However, it isn't clear to me that the proposed solution works.  For example, over the last few days I've had a track paused in Google Play Music which has been wrongly scrobbled many times.  During that time I have been listening to some podcasts in DoggCatcher.  So my last.fm profile looks like I have been alternating between podcasts (which I did play) and the paused song (which I did not).

The complication is, both Google Play Music and Doggcatcher play audio through the Android Music Player, and this is what SLS sees - ie, despite the fact I have tracks coming from two different apps, the "Enabled Apps" config in SLS only lists one.

I don't think that ignoring the last played song will fix the problem if SLS sees correct scrobbles from a different app in between the phantom scrobbles (unless SLS can also find out which program is driving Android Music Player.)
